### PR TITLE
Truncate microseconds to scale of 6

### PR DESCRIPTION
**Pull request by kevinkoltz**
Microsoft SQL Server defaults to 7, but Elixir NaiveDateTime supports up to 6

@mjaric I know you have a WIP branch for the encoding/decoding, so not sure if you want to pull this or not.  I added a test with an example datetime2 having a scale of 7, which was giving me NaiveDateTime errors with Elixir 1.8.2.  I have not tested with 1.9+.

Example error:
```
iex(2)> NaiveDateTime.from_erl!({{2000, 1, 1}, {13, 30, 15}}, {123_456, 6}) 
~N[2000-01-01 13:30:15.123456]
iex(3)> NaiveDateTime.from_erl!({{2000, 1, 1}, {13, 30, 15}}, {1_234_567, 7})
** (ArgumentError) cannot convert {{2000, 1, 1}, {13, 30, 15}} to naive datetime, reason: :invalid_time
    (elixir) lib/calendar/naive_datetime.ex:764: NaiveDateTime.from_erl!/3
```

### CODE REVIEW

File: test/datetime_test.exs [hunk 1]
```diff
@@ -165,15 +173,46 @@ defmodule DatetimeTest do
...
+    end
+
+    test "with scale of 6" do
+      enc = Types.encode_datetime2(@datetime_fsec)
+      assert Types.decode_datetime2(7, enc) == ~N[2015-04-08 15:16:23.123456]
+    end
+
+    test "with scale of 0" do
+      enc = Types.encode_datetime2({@date, {13, 56, 23, 0}}, 0)
+      assert Types.decode_datetime2(0, enc) == ~N[2015-04-08 13:56:23]
```
Comment (mjaric) [inline test/datetime_test.exs]: I think this is not what TDS should return, instead of decoding datetime to elixir NaiveDateTime or DateTime we should still keep TDS more generic and keep only tuples. Then in ecto 2.x, tds_ecto  or ecto_sql mssql adapter there should be `loader` and `dumper` that should convert from and to tuple to `:naive_datetime`, `:naive_datetime_usec`... so ecto can handle them. 

If we do this, then we can introduce Tds.DateTime, Tds.DateTime2 and Tds.DateTimeOffset as custom ecto types to add support for 7th digit precision for users that need it. This also should be case for migrations, if column is defined in migration as :navie_datetime_usec without precision option it should default to 6. If someone need 7 digits, then it should be explicit setup in migration file.

Do you agree?

File: lib/tds/types.ex [hunk 2]
```diff
@@ -1708,8 +1713,11 @@ defmodule Tds.Types do
         <<fsec::little-unsigned-40>>
     end
   end
+
   def encode_time(time), do: encode_time(time, @max_time_scale)
+
   # DateTime2
+  @spec decode_datetime2(number, binary) :: NaiveDateTime.t()
```
Comment (mjaric) [inline lib/tds/types.ex]: As mentioned in my  comment below (test case) decoder should return tuple so we can support 7 digit precision in case of custom ecto datetime type

File: lib/tds/types.ex [hunk 3]
```diff
@@ -1585,13 +1588,14 @@ defmodule Tds.Types do
...
     min = trunc(mins - hour * 60)
+
     {date, {hour, min, 0}}
     |> NaiveDateTime.from_erl!()
   end
 
   def encode_smalldatetime(nil), do: nil
 
-  def encode_smalldatetime(%{calendar: _}=naive_datetime) do
+  def encode_smalldatetime(%{calendar: _} = naive_datetime) do
```
Comment (mjaric) [inline lib/tds/types.ex]: We can keep this, but if ecto mssql adapter is well covered with edge cases +  default loaders and dumpers handle case for naivedatetime, this should be sufficient except for those who use tds directly.

File: test/datetime_test.exs (hunk 1 above)
Comment (kevinkoltz) [reply test/datetime_test.exs]: Yeah, this makes sense to keep the datetime's normalized as tuples and let the Ecto library, Tds.DateTime2, other custom Ecto types handle the final Ecto type encoding/decoding.  I had a custom type for Datetime2, which did the truncating and can help out with that.
